cpp/src/join/semi_join.cu

```diff
@@ -92,12 +92,22 @@ std::unique_ptr<rmm::device_uvector<cudf::size_type>> left_semi_anti_join(
                                                 equality_build);
   auto hash_table     = *hash_table_ptr;
 
-  thrust::for_each_n(rmm::exec_policy(stream),
-                     thrust::make_counting_iterator<size_type>(0),
-                     right_num_rows,
-                     [hash_table] __device__(size_type idx) mutable {
-                       hash_table.insert(thrust::make_pair(idx, true));
-                     });
+  // if compare_nulls == NOT_EQUAL, we can simply ignore any rows that are
+  // entirely NULL as they will never compare to equal.
```

Comment:
This isn't exactly true is it? The row needs a single null entry to skip the hash table. It isn't required that the row is entirely null.

I have a feeling this is a copy/paste thing and the original comment is incorrect as well.